### PR TITLE
FINERACT-1838: Loan account external events with repayment schedule

### PR DESCRIPTION
**Pull request by josehernandezfintecheandomx**
## Description

The Loan Account Repayment schedule is not included in external event when approving or disburse a Loan

[FINERACT-1838](https://issues.apache.org/jira/browse/FINERACT-1838).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by galovics** [comment]
@josehernandezfintecheandomx you can use https://issues.apache.org/jira/browse/FINERACT-1734 as a JIRA ticket for the PR title and the commit

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/serializer/loan/LoanBusinessEventSerializer.java [hunk 1]
```diff
@@ -46,6 +48,9 @@ public <T> boolean canSerialize(BusinessEvent<T> event) {
     protected <T> ByteBufferSerializable toAvroDTO(BusinessEvent<T> rawEvent) {
         LoanBusinessEvent event = (LoanBusinessEvent) rawEvent;
         LoanAccountData data = service.retrieveOne(event.get().getId());
+        if (event instanceof LoanApprovedBusinessEvent || event instanceof LoanDisbursalBusinessEvent) {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/serializer/loan/LoanBusinessEventSerializer.java]: This is not gonna fly. Either let's create a specific Serializer for these 2 events and mark it as priority. An example for that is `LoanChargeDeletedBusinessEventSerializer`

Or, rather fetch all the data for every LoanBusinessEvent (including the repayment schedule data).

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/serializer/loan/LoanBusinessEventSerializer.java (hunk 1 above)
Comment (josehernandezfintecheandomx) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/serializer/loan/LoanBusinessEventSerializer.java]: Done, It was set for every LoanBusinessEvent

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/serializer/loan/LoanBusinessEventSerializer.java [hunk 2]
```diff
@@ -46,6 +46,7 @@ public <T> boolean canSerialize(BusinessEvent<T> event) {
     protected <T> ByteBufferSerializable toAvroDTO(BusinessEvent<T> rawEvent) {
         LoanBusinessEvent event = (LoanBusinessEvent) rawEvent;
         LoanAccountData data = service.retrieveOne(event.get().getId());
+        data = service.fetchRepaymentScheduleData(data);
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/serializer/loan/LoanBusinessEventSerializer.java:49]: @josehernandezfintecheandomx do you think we can get this data with the retrieveOne call directly (so like extending the native query)?

**Review by galovics — APPROVED** [review]

**Pull request merged**